driver/src/contracts/stablex_contract.rs

```diff
@@ -51,7 +46,17 @@ impl BatchExchange {
 #[cfg_attr(test, automock)]
 pub trait StableXContract {
     fn get_current_auction_index(&self) -> Result<U256>;
-    fn get_auction_data(&self, index: U256) -> Result<(AccountState, Vec<Order>)>;
+    /// Retrieve one batch of auction data.
+    /// `block` is needed because the state of the smart contract could change
+    /// between blocks which would make the returned auction data inconsistent
+    /// between calls.
+    fn get_auction_data_batched(
```

Comment:
I might suggest calling this `get_paginated_auction_data` or get `get_auction_data_paginated`